addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -78,16 +90,118 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+        return this.getEvohomeBridge().getZoneSchedule("temperatureZone", this.getId());
+    }
+
+    private LocalDateTime getNextSetPointChange() {
+        return getNextSetPointChange(this.getDailySchedules());
+    }
+
+    private LocalDateTime getNextSetPointChange(DailySchedules schedule) {
+        if (schedule != null) {
+            int currentWeekDay = LocalDateTime.now().getDayOfWeek().getValue() - 1;
+            logger.debug("Current Weekday: " + currentWeekDay);
+            LocalDateTime nextSetPointChange = null;
+            for (DailySchedule daily : schedule.getSchedules()) {
+                // handle wrap around
+                int dayDelta = 0;
+                if (daily.getWeekday() < currentWeekDay) {
+                    dayDelta = daily.getWeekday() + 7 - currentWeekDay;
+                } else {
+                    dayDelta = daily.getWeekday() - currentWeekDay;
+                }
+                LocalDate setPointDay = LocalDate.now().plusDays(dayDelta);
+                for (Switchpoint sp : daily.getSwitchpoints()) {
+                    LocalDateTime spNextOccurance = LocalDateTime.of(setPointDay,
+                            LocalTime.parse(sp.getTimeOfDay().toString()));
+                    if (spNextOccurance.isAfter(LocalDateTime.now())) {
+                        // setPoint is after now we are interested in it.
+                        if (nextSetPointChange == null || nextSetPointChange.isAfter(spNextOccurance)) {
+                            nextSetPointChange = spNextOccurance;
+                        }
                     }
                 }
             }
+            if (nextSetPointChange == null) {
+                return LocalDateTime.now().plusMinutes(this.getOverrideTime());
+            }
+            return nextSetPointChange;
+        } else {
+            return LocalDateTime.now().plusMinutes(this.getOverrideTime());
+        }
+    }
+
+    private double getCurrentSetPoint() {
+        return getCurrentSetPoint(this.getDailySchedules());
+    }
+
+    private double getCurrentSetPoint(DailySchedules schedule) {
+        if (schedule != null) {
+            int currentWeekDay = LocalDateTime.now().getDayOfWeek().getValue() - 1;
+            logger.debug("Current Weekday: " + currentWeekDay);
+            LocalDateTime previousSetPointChange = null;
+            double previousSetPoint = 0;
+            for (DailySchedule daily : schedule.getSchedules()) {
+                // handle wrap around backwards
+                int dayDelta = 0;
+                if (daily.getWeekday() > currentWeekDay) {
+                    dayDelta = daily.getWeekday() - 7 + currentWeekDay;
+                } else {
+                    dayDelta = daily.getWeekday() - currentWeekDay;
+                }
+                logger.debug("Day Delta: " + dayDelta);
+                LocalDate setPointDay = LocalDate.now().plusDays(dayDelta);
+                logger.debug("Local Date: " + setPointDay);
```

Comment:
.. and this one.


I think it definitely makes sense to make this available in logging, though it needs a bit more effort to make it usable for other people as well :-)

Reply:
I thought about this, but don't think it's really likely to be used in logging.